src/init/features/functions/golang.ts

```diff
@@ -0,0 +1,113 @@
+const SDK_PATH = "github.com/FirebaseExtended/firebase-functions-go";
+
+async function init(setup: unknown, config: Config) {
+  await writeModFile(config);
+  await config.askWriteProjectFile("functions/functions.go", MAIN_TEMPLATE);
+  await config.askWriteProjectFile("functions/.gitignore", GITIGNORE_TEMPLATE);
+  utils.logLabeledBullet(
+    "functions",
+    "Welcome to the preview version of the Firebase Functions Go SDK. " +
+      "Please be a respectful member of the community and keep this a secret. " +
+      "Until the GitHub repo is public, we have set up a vendored module for you. " +
+      "To update to the latest preview of the SDK, download " +
+      SDK_DROP +
+      " and extract its contents into your " +
+      clc.bold("firebase-functions-go") +
+      " directory"
+  );
+}
+
+// writeModFile is meant to look like askWriteProjectFile but it generates the contents
+// dynamically using the go tool
+async function writeModFile(config: Config) {
+  const modPath = config.path("functions/go.mod");
+  if (fs.existsSync(modPath)) {
+    const shoudlWriteModFile = await promptOnce({
+      type: "confirm",
+      message: "File " + clc.underline("functions/go.mod") + " already exists. Overwrite?",
+      default: false,
+    });
+    if (!shoudlWriteModFile) {
+      return;
+    }
+
+    // Go will refuse to overwrite an existing mod file.
+    fs.unlinkSync(modPath);
+  }
+
+  // Nit(inlined) can we look at functions code and see if there's a domain mapping?
+  const modName = await promptOnce({
+    type: "input",
+    message: "What would you like to name your module?",
+    default: "acme.com/functions",
+  });
+
+  // Manually create a go mod file because (A) it's easier this way and (B) it seems to be the only
+  // way to set the min Go version to anything but what the user has installed.
+  config.writeProjectFile(
+    "functions/go.mod",
+    "module " +
+      modName +
+      "\n\ngo " +
+      RUNTIME_VERSION +
+      "\n\n" +
+      "require " +
+      SDK_PATH +
+      " v0.0.0\n\n" +
+      "replace " +
+      SDK_PATH +
+      " => ./firebase-functions-go\n"
+  );
+
+  const download = await fetch(SDK_DROP);
+  if (!download.body) {
+    logger.debug("Unexpected empty body response when downloading firebase-functions-go SDK");
+    throw new FirebaseError("Failed to download firebase-functions-go SDK");
+  }
+  if (!download.ok) {
+    throw new FirebaseError("Faield to download firebase-functions-go SDK");
+  }
+  // ESLint doesn't let us have a (non-constructor) function that starts with a captial letter,
+  // but this isn't our function to rename.
+  // eslint-disable-next-line new-cap
+  const extractArchive = unzipper.Extract({ path: config.path("functions/firebase-functions-go") });
+  await promisify(stream.pipeline)(download.body, extractArchive);
+
+  // Should this come later as "would you like to install dependencies" to mirror Node?
```

Comment:
I _think_ it makes sense to just always `go get` instead of prompting like we do for node. If I was developing, I'd prefer that, and I asked a few other peeps who have developed in go which they would expect, and they agreed.

Reply:
Sounds good. Removed comment.